### PR TITLE
Bifurcate DebtOrder into LoanRequest and Loan

### PR DESCRIPTION
**Pull request by saturnial**

### CODE REVIEW

File: src/loan/loan_request.ts [hunk 1]
```diff
@@ -0,0 +1,324 @@
...
+        );
+
+        const interestRate = InterestRate.fromRaw(loanOrder.interestRate);
+
+        const termLength = new TimeInterval(
+            loanOrder.termLength.toNumber(),
+            loanOrder.amortizationUnit,
+        );
+
+        const debtorAddress = new EthereumAddress(loanOrder.debtor!); // TODO(kayvon): this could throw.
```
Comment (graemecode) [inline src/loan/loan_request.ts]: What needs to be done for this TODO?

File: __test__/unit/loan/loan.spec.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+// Internal dependencies
+import { Dharma } from "../../../src/dharma";
+
+jest.unmock("@dharmaprotocol/contracts");
+
+import { IS_EXPIRED_SCENARIOS } from "./scenarios/is_expired_scenarios";
+import { DEBT_ORDER_PARAMS_ONE } from "./scenarios/valid_debt_order_params";
+
+// Test runners
+import { testCancel } from "./runners/cancel_as_debtor";
```
Comment (saturnial) [inline __test__/unit/loan/loan.spec.ts]: Plz clean up these imports

File: __test__/unit/loan/loan.spec.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
+// Internal dependencies
```
Comment (saturnial) [inline __test__/unit/loan/loan.spec.ts]: I would also move this to a dir called integration

File: __test__/unit/loan/loan.spec.ts (hunk 2 above)
Comment (chrismin) [reply __test__/unit/loan/loan.spec.ts]: Done.

File: __test__/unit/loan/loan.spec.ts (hunk 3 above)
Comment (chrismin) [reply __test__/unit/loan/loan.spec.ts]: Done.

File: __test__/integration/loan/loan.spec.ts [hunk 4]
```diff
@@ -0,0 +1,18 @@
+// Internal dependencies
+import { Dharma } from "../../../src/dharma";
+
+jest.unmock("@dharmaprotocol/contracts");
+
+import { IS_EXPIRED_SCENARIOS } from "./scenarios/is_expired_scenarios";
+import { DEBT_ORDER_PARAMS_ONE } from "./scenarios/valid_debt_order_params";
```
Comment (saturnial) [inline __test__/integration/loan/loan.spec.ts]: we should rename this constant

File: src/loan/loan_request.ts (hunk 1 above)
Comment (saturnial) [reply src/loan/loan_request.ts]: loan orders shouldn't hold onto optional properties. we can address this down the line.

**Pull request merged**